ivy/functional/ivy/losses.py

```diff
@@ -183,13 +198,51 @@ def binary_cross_entropy(
     ivy.array([0.223, 0.223, 0.223, 0.223])
     """
     ivy.utils.assertions.check_elem_in_list(reduction, ["none", "sum", "mean"])
-    pred = ivy.clip(pred, epsilon, 1 - epsilon)
-    return _reduce_loss(
-        reduction,
-        ivy.add(ivy.log(pred) * true, ivy.log(1 - pred) * (1 - true)),
-        None,
-        out,
-    )
+
+    if not (0.0 <= epsilon <= 1.0):
+        raise ValueError("epsilon should be a float in [0, 1]")
+
+    if not from_logits and pos_weight is not None:
+        raise ValueError("pos_weight is only allowed when from_logits is set to True")
+
+    true = true.astype(pred.dtype)
+
+    true = true.astype(pred.dtype)
+    epsilon = ivy.asarray(epsilon, dtype=pred.dtype)
+
+    true = true * (1.0 - epsilon) + 0.5 * epsilon
+
+    if from_logits:
+        if pos_weight is not None:
+            num_classes = pred.shape[0] if len(pred.shape) == 1 else pred.shape[1]
+            if pos_weight.shape[0] != num_classes:
+                raise ValueError(
+                    "pos_weight must have the same size as the number of classes in"
+                    " pred at non-singleton dimension 1"
+                )
+            epsilon_ = 1e-7
```

Comment:
Actually, the current version of `ivy.binary_cross_entropy()` used `epsilon` as an alias for `label-smoothing`.  Thus, I let it be called `epsilon` instead of changing it to `label-smoothing`.

The `epsilon_` is different from epsilon  and  is taken from TensorFlow implementation of the [same function](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/backend.py#L5035) and it values is 
```python
# Epsilon fuzz factor used throughout the codebase.
_EPSILON = 1e-7
```
defined at `/opt/fw/tensorflow/keras/backend_config.py`  and they've added this small value of ` epsilon_`  to the `output` and `1-output` when computing the logarithm to avoid taking the logarithm of `0`. This is because the logarithm of 0 is undefined and would result in an `error`. Adding a small value of epsilon ensures that the argument of the logarithm is always greater than 0, avoiding this issue.

Thus, the whole  source of confusion is because  we are using epsilon as an alias for label-smoothing!